### PR TITLE
Extract username from session

### LINKED ISSUE #274: AuthenticatedUser.get_name() now returns an empty string
### What did you do?

One of my applications started experiencing 500s of:

> Operation failed - Most likely the backend service failed. Please try again.

So I upgraded from `1.9.0` to `2.3.0` assuming there must have been a change to the Last.fm API that broke my deployed application.

```python
>>> network = pylast.LastFMNetwork(
    api_key=LASTFM_API_KEY,
    api_secret=LASTFM_API_SECRET,
    token=TOKEN_GENERATED_BY_LASTFM,
)
>>> user = network.get_authenticated_user()
>>> user.get_friends()[0].name
'janedoe'
>>> user.name
''
>>> user.get_name()
''
```

### What did you expect to happen?

`AuthenticatedUser.get_name()` used to return the username.

### What actually happened?

`AuthenticatedUser.get_name()` now returns an emptry string.

### What versions of OS, Python and pylast are you using?

- macOS 10.13.5
- Python 3.6.3
- pylast 2.3.0

### PR DESCRIPTION
**Pull request by jacebrowning**
Fixes #274

### CODE REVIEW

**Comment by jacebrowning** [comment]
@hugovk With this change, my application is now able to use the old behavior:

```python
 network = pylast.LastFMNetwork(
    api_key=settings.LASTFM_API_KEY,
    api_secret=settings.LASTFM_API_SECRET,
    token=token,
)
user = network.get_authenticated_user()
username = user.get_name()
print(username)
```

Any feedback on this implementation?

**Comment by hugovk** [comment]
Thanks for the PR!

---

First of all, this changes the public API. 

`get_web_auth_session_key` used to return a single `session_key` value, but now returns a `session_key, username` tuple.

There's quite a bit of [existing third-party code](https://github.com/search?q=get_web_auth_session_key&type=Code) out there that is expecting a single value, and this would break that code.

How about adding a new method, something like this?

```python
    def get_web_auth_session_key_and_username(self, url, token=""):
        # as get_web_auth_session_key is now in the PR
        # ...
        return session_key, username

    def get_web_auth_session_key(self, url, token=""):
        session_key, username = get_web_auth_session_key_and_username(url, token)
        return session_key
```

Then the existing code out there can still call `get_web_auth_session_key`.

---

Please could you also fix the lint formatting check? You can do it automatically with:
```bash
pip install black
black .
```

File: setup.cfg [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 [flake8]
-max_line_length = 88
+ignore = E501
 
 [metadata]
 license_file = COPYING
 
 [pycodestyle]
-max_line_length = 88
+ignore = E501
```
[the remark this answers is not in the document]
Comment (hugovk) [reply setup.cfg]: W503 should be ignored, but let's keep `max_line_length = 88` and remove E501 from the ignore.

Instead, for the first one, we can either append `# noqa E501`:
```diff
-            self.session_key, self.username = sk_gen.get_web_auth_session_key_and_username(
+            self.session_key, self.username = sk_gen.get_web_auth_session_key_and_username(  # noqa: E501
```

Or rename it to something shorter, maybe remove the `and` like `get_web_auth_session_key_username`.

For the second one, let's split docstring over two lines.

[the remark this answers is not in the document]
Comment (jacebrowning) [reply setup.cfg]: OK. I'll disable per line.

**Comment by hugovk** [comment]
Thank you!

**Comment by hugovk** [comment]
This had been released in pylast 3.1.0. Thanks again!

**Comment by Qwerty-Space** [comment]
> This had been released in pylast 3.1.0. Thanks again!

I'm still having the issue in 3.1.0

**Comment by Qwerty-Space** [comment]
> This had been released in pylast 3.1.0. Thanks again!

@hugovk I'm still having this issue in 3.1.0

**Pull request merged**